pkg/cmd/deployment/deployment.go

```diff
@@ -0,0 +1,26 @@
+package deployment
+
+import (
+	cmdCreate "github.com/cli/cli/v2/pkg/cmd/deployment/create"
+	cmdDelete "github.com/cli/cli/v2/pkg/cmd/deployment/delete"
+	cmdList "github.com/cli/cli/v2/pkg/cmd/deployment/list"
+	cmdUpdate "github.com/cli/cli/v2/pkg/cmd/deployment/update"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/spf13/cobra"
+)
+
+func NewCmdDeployment(f *cmdutil.Factory) *cobra.Command {
```

Comment:
Fair questions, and I was wondering about a sub-command to align to the Deployment API and it's various endpoints.

Though I've sort of figured the various endpoints are a bit clunky, for my use cases, and in replicating them as-is in the CLI it'd be surfacing that complexity to users rather than simplifying it. Similar to how GitHub Actions seems to radically simplify the usage of the Deployments API too.

This thinking comes from reading about the intent of this CLI to be a bit opinionated about workflows and simplify the experience[^1].

[^1]: https://github.com/cli/cli/blob/trunk/docs/gh-vs-hub.md

Reply:
> Though I've sort of figured the various endpoints are a bit clunky, for my use cases, and in replicating them as-is in the CLI it'd be surfacing that complexity to users rather than simplifying it.

100%. I think the question is "how complex is it going to be if people want that as first class behaviour". Are we going to tell them to use `gh api`? Are we going to end up having some mixing of abstraction layers in the commands? Not sure yet.

Reply:
Also worth highlighting `gh environment` already exists as a command for another use-case if the line of thinking is towards more top-level commands e.g. `gh deployment-status create` or the like.

Reply:
@sjparkinson @williammartin : I'd like to propose pivoting back to the issue in order to discuss some thoughts about what deployment support might look like with an eye to GitHub UI experience.  I think this PR is a good start, but there are pieces of the experience missing that I'd prefer capturing with screenshots and aligning with others feedback.